src/components/ViewRequest/HiringPanel.tsx

```diff
@@ -7,15 +8,20 @@ import { useParams } from "react-router-dom";
 
 const HiringPanel = () => {
   const params = useParams();
-  const request = useRequest(Number(params.requestId));
+  const requestId = Number(params.requestId);
+  const request = useRequest(requestId);
   const postRequest = usePostRequest();
+  const addNote = useAddNote(requestId);
 
   const onChange = useCallback(
     (ev: ChangeEvent<HTMLInputElement>) => {
       postRequest.mutate({
-        requestId: Number(params.requestId),
+        requestId: requestId,
         postRequest: { panelRequired: ev.currentTarget.checked ? "Yes" : "No" },
       });
+      addNote.mutate(
```

Comment:
Add addNote.isLoading to the disabled attribute in line 64?  There is a bit of lag and a flash of blue of it becoming enabled but still "Yes" before becoming "No".  This would likely help with that.